### PR TITLE
945: Upgrade and simplify build, support HMR

### PR DESCRIPTION
**Pull request by kadamwhite**
We encountered a number of issues on #5 that were caused by our build setup. This PR introduces the latest version of our `wp-scripts` adapter layer, which provides more WP-native asset building and loading without losing our hot-reloading abilities.

### CODE REVIEW

**Comment by kadamwhite** [comment]
@goldenapples Requesting a retroactive review from you, but merging for now so I can use this elsewhere

File: node_modules/serve-index/public/style.css [hunk 1]
```diff
@@ -0,0 +1,257 @@
+* {
```
Comment (goldenapples) [inline node_modules/serve-index/public/style.css:1]: Should this have been checked in? I thought the node_modules directory was git ignored.

File: src/block.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+	"description": "This is a bit of a hack: using a bare block.json lets us trick wp-scripts into building our frontend and editor bundles",
```
Comment (goldenapples) [inline src/block.json:2]: sneaky! 

I think you can also add entry paths to wp-scripts via a command line argument, but this works.

File: src/blocks/example-dynamic/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	/**
+	 * @see ./edit.js
+	 */
+	edit: Edit,
+} );
+
+// Block HMR boilerplate.
+if ( module.hot ) {
+	module.hot.accept();
+	const { deregister, refresh } = require( '../../helpers/hot-blocks.js' );
```
Comment (goldenapples) [inline src/blocks/example-dynamic/index.js:38]: This seems like a nice improvement. Does this ensure that the HMR code isn't built into the production bundle?

**Review by goldenapples — COMMENTED** [review]
Looks good, thanks for updating this. If and when you get around to writing up the whole setup, I'd love to see a prepackaged version of this setup that can be easily used to start new projects.

File: src/blocks/example-dynamic/index.js (hunk 3 above)
Comment (kadamwhite) [reply src/blocks/example-dynamic/index.js:38]: It does, yes!

File: src/block.json (hunk 2 above)
Comment (kadamwhite) [reply src/block.json:2]: Tried that and it wasn't working as I'd expect, but didn't try very hard. Will have a go at it again next project :)

File: node_modules/serve-index/public/style.css (hunk 1 above)
Comment (kadamwhite) [reply node_modules/serve-index/public/style.css:1]: It should not be, I don't know how this happened. I'll revert it.

**Pull request merged**